### ISSUE TITLE
Redmine 4.0.4

### ISSUE THREAD

**Issue by chmich** [ozguryazilimas/redmine_default_assignee#6]
Dear Özgür,

i liked your plugin very much, but after updating to redmine-4.0.4:

after..
```
$ RAILS_ENV=production bundle exec rake redmine:plugins:migrate
```
error:
```
(in /srv/redmine/redmine-4.0.4)
rake aborted!
NoMethodError: undefined method `alias_method_chain' for SettingsController:Class
Did you mean?  alias_method
/srv/redmine/redmine-4.0.4/plugins/redmine_default_assignee/lib/redmine_default_assignee/patches/settings_controller_patch.rb:10:in `block in included'
/srv/redmine/redmine-4.0.4/plugins/redmine_default_assignee/lib/redmine_default_assignee/patches/settings_controller_patch.rb:9:in `class_eval'
/srv/redmine/redmine-4.0.4/plugins/redmine_default_assignee/lib/redmine_default_assignee/patches/settings_controller_patch.rb:9:in `included'
/srv/redmine/redmine-4.0.4/plugins/redmine_default_assignee/init.rb:32:in `include'
/srv/redmine/redmine-4.0.4/plugins/redmine_default_assignee/init.rb:32:in `block (2 levels) in <top (required)>'
/srv/redmine/redmine-4.0.4/plugins/redmine_default_assignee/init.rb:30:in `each'
/srv/redmine/redmine-4.0.4/plugins/redmine_default_assignee/init.rb:30:in `block in <top (required)>'
/home/redmine/.rvm/gems/ruby-2.4.1/gems/activesupport-5.2.3/lib/active_support/callbacks.rb:426:in `instance_exec'
/home/redmine/.rvm/gems/ruby-2.4.1/gems/activesupport-5.2.3/lib/active_support/callbacks.rb:426:in `block in make_lambda'
/home/redmine/.rvm/gems/ruby-2.4.1/gems/activesupport-5.2.3/lib/active_support/callbacks.rb:198:in `block (2 levels) in halting'
/home/redmine/.rvm/gems/ruby-2.4.1/gems/activesupport-5.2.3/lib/active_support/callbacks.rb:606:in `block (2 levels) in default_terminator'
/home/redmine/.rvm/gems/ruby-2.4.1/gems/activesupport-5.2.3/lib/active_support/callbacks.rb:605:in `catch'
/home/redmine/.rvm/gems/ruby-2.4.1/gems/activesupport-5.2.3/lib/active_support/callbacks.rb:605:in `block in default_terminator'
/home/redmine/.rvm/gems/ruby-2.4.1/gems/activesupport-5.2.3/lib/active_support/callbacks.rb:199:in `block in halting'
/home/redmine/.rvm/gems/ruby-2.4.1/gems/activesupport-5.2.3/lib/active_support/callbacks.rb:513:in `block in invoke_before'
/home/redmine/.rvm/gems/ruby-2.4.1/gems/activesupport-5.2.3/lib/active_support/callbacks.rb:513:in `each'
/home/redmine/.rvm/gems/ruby-2.4.1/gems/activesupport-5.2.3/lib/active_support/callbacks.rb:513:in `invoke_before'
/home/redmine/.rvm/gems/ruby-2.4.1/gems/activesupport-5.2.3/lib/active_support/callbacks.rb:131:in `run_callbacks'
/home/redmine/.rvm/gems/ruby-2.4.1/gems/activesupport-5.2.3/lib/active_support/reloader.rb:89:in `prepare!'
/home/redmine/.rvm/gems/ruby-2.4.1/gems/railties-5.2.3/lib/rails/application/finisher.rb:63:in `block in <module:Finisher>'
/home/redmine/.rvm/gems/ruby-2.4.1/gems/railties-5.2.3/lib/rails/initializable.rb:32:in `instance_exec'
/home/redmine/.rvm/gems/ruby-2.4.1/gems/railties-5.2.3/lib/rails/initializable.rb:32:in `run'
/home/redmine/.rvm/gems/ruby-2.4.1/gems/railties-5.2.3/lib/rails/initializable.rb:61:in `block in run_initializers'
/home/redmine/.rvm/gems/ruby-2.4.1/gems/railties-5.2.3/lib/rails/initializable.rb:60:in `run_initializers'
/home/redmine/.rvm/gems/ruby-2.4.1/gems/railties-5.2.3/lib/rails/application.rb:361:in `initialize!'
/srv/redmine/redmine-4.0.4/config/environment.rb:14:in `<top (required)>'
/home/redmine/.rvm/gems/ruby-2.4.1/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:291:in `require'
/home/redmine/.rvm/gems/ruby-2.4.1/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:291:in `block in require'
/home/redmine/.rvm/gems/ruby-2.4.1/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:257:in `load_dependency'
/home/redmine/.rvm/gems/ruby-2.4.1/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:291:in `require'
/home/redmine/.rvm/gems/ruby-2.4.1/gems/railties-5.2.3/lib/rails/application.rb:337:in `require_environment!'
/home/redmine/.rvm/gems/ruby-2.4.1/gems/railties-5.2.3/lib/rails/application.rb:520:in `block in run_tasks_blocks'
/home/redmine/.rvm/gems/ruby-2.4.1/gems/rake-12.3.2/exe/rake:27:in `<top (required)>'
/home/redmine/.rvm/gems/ruby-2.4.1/bin/ruby_executable_hooks:15:in `eval'
/home/redmine/.rvm/gems/ruby-2.4.1/bin/ruby_executable_hooks:15:in `<main>'
Tasks: TOP => redmine:plugins:migrate => environment
(See full trace by running task with --trace)
```

**Comment by zaburt** [comment]
Hi, the plugin was not compatible with Redmine 4.x that is why the error you reported was happening. Master branch is ported to Redmine 4.x now. Please let us know if it works for you too.

**Comment by chmich** [comment]
Dear Özgür,

now i can install the plugin and see the UI in redmine.
But, after klick save-button,

![image](https://user-images.githubusercontent.com/31048265/61582756-1a9fbe80-ab2f-11e9-8adb-0ba2dbbf5b54.png)


i receipt this error:

```
Started POST "/projects/sedlmair/rda_project_settings" for 31.10.128.205 at 2019-07-20 20:40:12 +0200
Processing by ProjectsController#rda_project_settings as HTML
  Parameters: {"utf8"=>"✓", "authenticity_token"=>"/Rr6j5I6agVQemCqagpOtv0WeZN+VKZ4fJ8IYkmOtOMozHyLySftlOfwKO7iKEKobY6YKPDYdL816I0nezKd5w==", "settings"=>{"project_id"=>"5", "default_assignee"=>{"5"=>"", "1"=>"1", "7"=>"", "8"=>"", "9"=>"", "2"=>"", "4"=>"", "3"=>"", "14"=>"", "13"=>"", "10"=>"", "11"=>""}}, "commit"=>"Save", "id"=>"sedlmair"}
  Current user: christian (id=1)
Completed 500 Internal Server Error in 12ms (ActiveRecord: 5.4ms)
  
ActiveModel::ForbiddenAttributesError (ActiveModel::ForbiddenAttributesError):
  
plugins/redmine_default_assignee/lib/redmine_default_assignee/patches/projects_controller_patch.rb:25:in `rda_project_settings'
lib/redmine/sudo_mode.rb:63:in `sudo_mode'
Started GET "/time_loggers/render_menu?project_id=5&_=1563647837290" for 31.10.128.205 at 2019-07-20 20:40:17 +0200
Processing by TimeLoggersController#render_menu as */*
  Parameters: {"project_id"=>"5", "_"=>"1563647837290"}
  Current user: christian (id=1)
  Rendered plugins/time_logger/app/views/time_loggers/_embed_menu.html.erb (5.4ms)
Completed 200 OK in 17ms (Views: 5.3ms | ActiveRecord: 7.1ms)
```

**Comment by zaburt** [comment]
This needed some changes to handle strong params and new model attributes, should work fine now

**Comment by chmich** [comment]
Thanks, Özgür, now it works fine!
best regards,
Christian

**Comment by chmich** [comment]
Dear Özgür,

a little nice-to-have:

After creating a new subproject from a subproject, the selects are empty and you can choose nothing. reload doesn't help.

only after assign a issue to the prject, you have the regular selects and  you can select a assignee

![image](https://user-images.githubusercontent.com/31048265/61651934-3afa8500-acb7-11e9-8cb1-2b84677c9314.png)

**Comment by zaburt** [comment]
Please open different issues for different subjects. Reopening an issue for an unrelated subject makes it hard to maintain.

Also note that the issue you mention above is handled by Redmine, I suspect there isn't anything we can do about it as we show what Redmine provides us.
